lib/posix-process/process.c

```diff
@@ -342,8 +344,109 @@ UK_SYSCALL_R_DEFINE(int, setpriority, int, which, id_t, who, int, prio)
 	return rc;
 }
 
-int prctl(int option __unused, ...)
+UK_SYSCALL_R_DEFINE(int, prctl, int, option,
+		    unsigned long, arg2,
+		    unsigned long, arg3,
+		    unsigned long, arg4,
+		    unsigned long, arg5)
+{
+	UK_WARN_STUBBED();
+	return 0; /* syscall has no effect */
+}
+
+UK_LLSYSCALL_R_DEFINE(int, prlimit64, int, pid, unsigned int, resource,
+		      struct rlimit *, new_limit, struct rlimit *, old_limit)
+{
+	if (unlikely(pid != 0))
+		uk_pr_debug("Do not support prlimit64 on PID %u, use current process\n", pid);
+
+	/*
+	 * Lookup if resource can be set/retrieved
+	 */
+	switch(resource) {
+		case RLIMIT_STACK:
+			break;
+#if CONFIG_LIBVFSCORE
+		case RLIMIT_NOFILE:
+			break;
+#endif
+		default:
+			uk_pr_err("Unsupported resource %u\n",
+				  resource);
+			return -EINVAL;
+	}
+
+	/*
+	 * Set resource
+	 */
+	if (new_limit) {
+		switch(resource) {
+		default:
+			uk_pr_err("Ignore updating resource %u: cur = %llu, max = %llu\n",
+				  resource,
+				  (unsigned long long) new_limit->rlim_cur,
+				  (unsigned long long) new_limit->rlim_max);
+			break;
+		}
+	}
+
+	/*
+	 * Get resource
+	 */
+	if (!old_limit)
+		return 0;
+	switch(resource) {
+		case RLIMIT_STACK:
+			old_limit->rlim_cur = __STACK_SIZE;
+			old_limit->rlim_max = __STACK_SIZE;
+			break;
+
+#if CONFIG_LIBVFSCORE
+		case RLIMIT_NOFILE:
+			old_limit->rlim_cur = FDTABLE_MAX_FILES;
+			old_limit->rlim_max = FDTABLE_MAX_FILES;
+			//old_limit->rlim_max = 0x100000; /* linux limit */
+			break;
+#endif
+
+		default:
+			break;
+	}
+
+	uk_pr_debug("Resource %u: cur = %llu, max = %llu\n",
+		    resource,
+		    (unsigned long long) old_limit->rlim_cur,
+		    (unsigned long long) old_limit->rlim_max);
+	return 0;
+}
+
```

Comment:
These `getrlimit`, `setrlimit` and `getrusage` seem to cause a compilation error with newlib due to what I think to be a conflict with the ones already [there](https://github.com/unikraft/lib-newlib/blob/9a7aa73ba7526e88c5c9636667e3e012f2042c81/resource.c#L41).

Reply:
True, the fix is to remove the implementation from `newlib`.